packages/widget/src/controllers/transfers/substrate/build.ts

```diff
@@ -0,0 +1,40 @@
+import { SubstrateAssetTransfer } from '@buildwithsygma/sygma-sdk-core/substrate';
+import type { ApiPromise } from '@polkadot/api';
+import { type FungibleTokenTransferController } from '../fungible-token-transfer';
+
+export async function buildSubstrateFungibleTransactions(
+  this: FungibleTokenTransferController
+): Promise<void> {
+  const address = this.walletContext.value?.substrateWallet?.signerAddress;
+  if (
+    !this.sourceNetwork ||
+    !this.destinationNetwork ||
+    !this.resourceAmount ||
+    !this.selectedResource ||
+    !this.destinatonAddress ||
+    !address
+  ) {
+    return;
+  }
+
+  const substrateTransfer = new SubstrateAssetTransfer();
+  await substrateTransfer.init(
+    this.walletContext.value?.substrateWallet?.substrateProvider as ApiPromise,
+    this.env
+  );
+
+  const transfer = await substrateTransfer.createFungibleTransfer(
+    address,
+    this.destinationNetwork.chainId,
+    this.destinatonAddress,
+    this.selectedResource.resourceId,
+    String(this.resourceAmount)
+  );
+
+  const fee = await substrateTransfer.getFee(transfer);
```

Comment:
can you see fee in a variable for UI to show?

Reply:
sure